aiosonic/connection.py

```diff
@@ -168,36 +169,42 @@ def block_until_read_chunks(self):
         """Check if keep alive."""
         self.blocked = True
 
-    def release(self) -> None:
+    async def release(self) -> None:
         """Release connection."""
-        self.connector.release(self)
         self.requests_count += 1
         # if keep False and blocked (by latest chunked response), close it.
         # server said to close it.
         if self.requests_count >= self.connector.conn_max_requests or (
             not self.keep and self.blocked
         ):
             self.blocked = False
-            self.close()
+            await self.close()
         # ensure unblock conn object after read
         self.blocked = False
+        self.connector.release(self)
 
-    def ensure_released(self):
+    async def ensure_released(self):
         """Ensure the connection is released."""
         if self.blocked:
-            if self.writer:
-                self.writer._transport.abort()
             self.blocked = False
```

Comment:
@sonic182 I tried to keep the commit clean and simple but that means I may have overlooked details too. For example, we're setting blocked to False here and then awaiting for Release method, which will not await on Connection.close() because blocked is set to False to begin with. Let me re-test my crawler without this premature unblocking and see if everything is OK. If things are OK we might be able to do without this.

Reply:
Tried it. I now get ```RuntimeError: unable to perform operation on <TCPTransport closed=True reading=False 0x559fd5f4b250>; the handler is closed```

Bummer.